quay/quay-operator
# Downstream Component Images (PROJQUAY-828)

**Issue:** https://issues.redhat.com/browse/PROJQUAY-828

**Changelog:** Add initial downstream release Kustomize overlay for managed component container images.

**Docs:** Ensure that users know how to create `redhat-pull-secret` ([documentation exists](https://access.redhat.com/solutions/3533201)).

**Testing:** 
1. Create `redhat-pull-secret` in same namespace as `QuayRegistry`.
2. Add the above pull secret name to list of `imagePullSecrets` for the `ServiceAccount` named `default` in your namespace so it can pull the private images.
3. Create zero-config `QuayRegistry`.

Expected:
- All containers pull successfully.

**Details:** Adds Kustomize overlays for an initial downstream release using official Red Hat container images for relevant managed components. These container images require a [Red Hat pull secret](https://access.redhat.com/solutions/3533201). 

Also switches to deploy CentOS-based Postgres image when running in upstream mode.

**Remaining Issues**:
- [x] Ensure `pg_trgm` extension created for `quay` database 
- [x] Clair database doesn't have `clair` DB being created
- [x] Switch to using CentOS Postgres image for upstream and confirm all env vars and scripts continue working
